pullrequests/longest_consecutive_sequence/step2.go

```diff
@@ -0,0 +1,32 @@
+//lint:file-ignore U1000 Ignore all unused code
+package longestconsecutivesequence
+
+/*
+他の人のPRを見て、「手でやるとして、{0, 1, 2, 3, 4, 5} だったとして、0 から確認が終わった後に 1 を確認しないじゃないですか」（https://github.com/t-ooka/leetcode/pull/7/files#r1665902244 ）というodaさんのコメントを発見し、確かにその通りだと思ったので下記のように一度連続する数に含めたものは削除するように変更した。
+実際にLeetCode上でもStep1の場合、５回ほど提出しても実行時間がどれも1800msほどだったのに対し、Step2の場合は90msぐらいだったので、実際にとても早くなることが実感できた。
+
+質問：どうやらこの問題はUnion FIndを使っても解くことができるようですが、Union FIndを使って解く練習もするべきでしょうか？
+*/
+func longestConsecutiveStep2(nums []int) int {
+	numsMap := make(map[int]struct{}, len(nums))
+	for _, n := range nums {
+		numsMap[n] = struct{}{}
+	}
+	maxLength := 0
+	for n := range numsMap {
+		if _, ok := numsMap[n-1]; ok {
+			continue
+		}
+		length := 1
+		for {
+			n++
```

Comment:
このループの外の n を直接変化させているので、ループを抜けた後に n を扱う処理がある場合にバグが発生しやすいのでは、と思いました。（今回のケースではループを抜けた後には maxLength の更新だけなので問題なさそうです）

このループの前あたりに length と合わせてインクリメントする変数を宣言して n 自体とは区別するといいかもです。（適当に cur としましたが、これはいい変数名ではなさそうです）

```suggestion
                cur := n
		length := 1
		for {
			cur++
```

Reply:
`n+length`とするとn自体をインクリメントせずに表現できそうですね

Reply:
別の言い方をすると、一文字変数は説明書きが長くなるような使い方をするべきではない、ということかなと思います。
for n := range numsMap
とあったら、普通は「numsMap のキー」と理解するし、それ以上の理解はしないので、間で別のものにすり替えられていると見落とす、ということです。